party_menu/party_menu.lua

```diff
@@ -0,0 +1,521 @@
+end
+
+local function get_hp(pos)
+    if hp_modes[options.hp_mode_index] == 'hide' then
+        return ''
+    elseif party[pos].hp == 0 and hp_modes[options.hp_mode_index] == 'missinghp' then
+        return 'mHP:' .. string.format('%-5.5s ', 'DEAD')
+    elseif party[pos].hp == 0 and hp_modes[options.hp_mode_index] == 'hp/hp_max' then
+        return 'HP:' .. string.format('%-9.9s ', 'DEAD')
+    elseif party[pos].hp == 0 then
+        return 'HP:' .. string.format('%4.4s ', 'DEAD')
+    elseif hp_modes[options.hp_mode_index] == 'missinghp' then
+        return 'mHP:-' .. string.format('%-4.4s ', math.floor(party[pos].hp * 100 / party[pos].hp_percent) - party[pos].hp)
+    elseif hp_modes[options.hp_mode_index] == 'hp/hp_max' then
+        return 'HP:' .. string.format('%4.4s', party[pos].hp) .. '/' .. string.format('%-4.4s ', math.floor(party[pos].hp * 100 / party[pos].hp_percent))
+    else
+        return 'HP:' .. string.format('%4.4s ', party[pos].hp)
+    end
+end
+
+local function get_hp_color(pos)
+    local color
+    if not options.dynamic_coloring then
+        color = 'white'
+    elseif party[pos].hp_percent >= 95 then
+        color = 'darkgreen'
+    elseif party[pos].hp_percent >= 75 then
+        color = 'limegreen'
+    elseif party[pos].hp_percent >= 50 then
+        color = 'yellow'
+    elseif party[pos].hp_percent >= 25 then
+        color = 'darkorange'
+    elseif party[pos].hp_percent >= 11 then
+        color = 'red'
+    elseif party[pos].hp_percent < 11 then
+        color = 'darkred'
+    end
+    return color
+end
+
+local function get_mp_color(pos)
+    local color
+    if not options.dynamic_coloring then
+        color = 'white'
+    elseif party[pos].mp_percent >= 95 then
+        color = 'darkgreen'
+    elseif party[pos].mp_percent >= 75 then
+        color = 'limegreen'
+    elseif party[pos].mp_percent >= 50 then
+        color = 'yellow'
+    elseif party[pos].mp_percent >= 25 then
+        color = 'darkorange'
+    elseif party[pos].mp_percent >= 11 then
+        color = 'red'
+    elseif party[pos].mp_percent < 11 then
+        color = 'darkred'
+    end
+    return color
+end
+
+local function get_mp(pos)
+    if mp_modes[options.mp_mode_index] == 'mp' then
+        return string.format('MP:%4.4s ', party[pos].mp)
+    elseif mp_modes[options.mp_mode_index] == 'mp/mp_max' then
+        return string.format('MP:%4.4s', party[pos].mp) .. '/' .. string.format('%-4.4s ', math.floor(party[pos].mp * 100 / party[pos].mp_percent))
+    else
+        return ''
+    end
+end
+
+local function get_tp(pos)
+    if options.tp_visible then
+        return string.format('TP:%4.4s ', party[pos].tp)
+    else
+        return ''
+    end
+end
+
+local function get_tp_color(pos)
+    if party[pos].tp < 1000 or not options.dynamic_coloring then
+        return 'white'
+    else
+        return 'dodgerblue'
+    end
+end
+
+local function get_debuff_string(pos)
+    if options.debuffs_visible then
+        debuffs = 'DB:'
```

Comment:
This should be `local`. I'm also not too keen on that, design wise. I wouldn't guess that `DB` stands for that. I think this can safely be omitted without loosing any important information. If someone sees "weakness" or "paralyzed" written there, they know what it means. It also opens the option to move on from debuffs and show buffs as well, see the comment below.